src/transfer/messages.rs

```diff
@@ -184,6 +184,12 @@ pub struct JobDetail {
     pub entries: Option<Vec<BString>>,
     pub max_fails: Option<JobTaskCount>,
     pub priority: tako::Priority,
+
+    // Date when job was submitted
+    pub submission_date: DateTime<Utc>,
+
+    // Time when job was completed or now if job is not completed
+    pub completion_date_or_now: DateTime<Utc>,
```

Comment:
`completion_date_or_now` is sent over the wire instead of something like `Option<DateTime<Utc>>` to use server time for `now`?

Reply:
To compute makespan of job/duration of tasks you basically always needs something like (completion_date_or_now - start) so I wanted to keeped as simple as possible and do not introduce two additional field.